lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -1296,73 +1284,58 @@ impl<'s> SegmentHolder {
+            // Fill buffer with points of current chunk, need at least 2 points to deduplicate
+            points.clear();
+            points.extend(group_iter);
+            if points.len() < 2 {
+                continue;
             }
 
-            if last_point_id_opt == Some(point_id) {
-                let last_segment_id = last_segment_id_opt.unwrap();
-
-                let point_version = locked_segments[&segment_id].point_version(point_id);
-                let last_point_version = if let Some(last_point_version) = last_point_version_opt {
-                    last_point_version
-                } else {
-                    let version = locked_segments[&last_segment_id].point_version(point_id);
-                    last_point_version_opt = Some(version);
-                    version
-                };
-
-                // choose newer version between point_id and last_point_id
-                if point_version < last_point_version {
-                    log::debug!("Selected point {point_id} in segment {segment_id} for deduplication (version {point_version:?} versus {last_point_version:?} in segment {last_segment_id})");
-
-                    points_to_remove
-                        .entry(segment_id)
-                        .or_default()
-                        .push(point_id);
-                } else {
-                    log::debug!("Selected point {point_id} in segment {last_segment_id} for deduplication (version {last_point_version:?} versus {point_version:?} in segment {segment_id})");
-
-                    points_to_remove
-                        .entry(last_segment_id)
-                        .or_default()
-                        .push(point_id);
+            // Attach point version to
+            for dedup_point in &mut points {
+                dedup_point.version =
+                    locked_segments[&dedup_point.segment_id].point_version(dedup_point.point_id);
+            }
 
-                    last_point_id_opt = Some(point_id);
-                    last_segment_id_opt = Some(segment_id);
-                    last_point_version_opt = Some(point_version);
-                }
-            } else {
-                last_point_id_opt = Some(point_id);
-                last_segment_id_opt = Some(segment_id);
-                last_point_version_opt = None;
+            // Sort points from highest to lowest version
+            points.sort_unstable_by(|a, b| b.version.cmp(&a.version));
+
+            // Keep the first point, remove all others which are older
+            for &DedupPoint {
+                segment_id,
+                point_id,
+                version,
+            } in &points[1..]
+            {
+                log::debug!(
+                    "Selected point {point_id} in segment {segment_id} for deduplication (version {version:?} versus {:?} in segment {})",
+                    points[0].version,
+                    points[0].segment_id,
+                );
```

Comment:
I see it was also `debug!` before, but... For a segment with lots of old versions, this can create a lot of noise, doesn't it? 

I'd consider using `trace!` instead 🤔

Reply:
Yes, the plan was to demote this again with <https://github.com/qdrant/qdrant/pull/5586> (as it was promoted to debug for chaos testing). But it looks like that one is already merged.

I'll create a separate PR instead to demote theses, so it doesn't depend on these changes.

Reply:
https://github.com/qdrant/qdrant/pull/5604